### ISSUE TITLE
Document Infrastructure Requirements for testnet (and mainnet)

### ISSUE THREAD

**Issue by D4nte** [comit-network/comit.network#42]
## Problem

There is no clear explanation of what is the infrastructure needed to run COMIT once a developer is ready to go on testnet or mainnet.

## Goal

Ensure that users and developers are aware of what is needed to run COMIT on testnet or mainnet.

## Recommendation

A documentation that explains the different interactions between the components (cnd, blockchain nodes, wallets).

**Comment by yosriady** [comment]
Yes, I think this will also help developers understand what is needed to bring their app to production, beyond running nodes locally.

**Comment by da-kami** [comment]
Slides that should be embedded for giving the infrastructure introduction:

https://docs.google.com/presentation/d/e/2PACX-1vSnucTvT_KdrvxWnMf2WCbEegUsr6p9U0K9tT9i-zG-5Zu2JWKNcTymqW_lDFCTvgqLD_WncUMgY0zd/pub?start=false&loop=false&delayms=3000

Embed code:

<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vSnucTvT_KdrvxWnMf2WCbEegUsr6p9U0K9tT9i-zG-5Zu2JWKNcTymqW_lDFCTvgqLD_WncUMgY0zd/embed?start=false&loop=false&delayms=3000" frameborder="0" width="801" height="480" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>

We can of course modify them if needed, just ping me.

**Comment by da-kami** [comment]
Should this include how to configure 2x cnd nodes in a local environment or are we assuming that `comit-scripts` should be used? 

I think we need several different sections when it comes to "infrastructure":

1. Component interaction and setup when using `comit-scripts`
2. Component interaction and setup when running a local setup without `comit-scripts`
3. Component interaction and setup when running nodes on different machines (with testnet/mainnet nodes)

@D4nte I'd suggest to focus on 1. (and maybe 2.) in this ticket.

**Comment by yosriady** [comment]
Developers using `comit-scripts` to develop locally wouldn't need to think about infrastructure. 

So I think it's more for the next steps of going without `comit-scripts`, to deploying and launching their applications to a real testnet or mainnet.

**Comment by da-kami** [comment]
> Developers using `comit-scripts` to develop locally wouldn't need to think about infrastructure.
> 
> So I think it's more for the next steps of going without `comit-scripts`, to deploying and launching their applications to a real testnet or mainnet.

Do I understand you correctly @yosriady, you are saying that the component interaction is actually not important when using `comit-scripts`? 

I wonder if there should be a more general overview and then one that explains what `comit-scripts` does for you. Would that make sense from your point of view?

**Comment by yosriady** [comment]
> > Developers using `comit-scripts` to develop locally wouldn't need to think about infrastructure.
> > So I think it's more for the next steps of going without `comit-scripts`, to deploying and launching their applications to a real testnet or mainnet.
> 
> Do I understand you correctly @yosriady, you are saying that the component interaction is actually not important when using `comit-scripts`?
> 
> I wonder if there should be a more general overview and then one that explains what `comit-scripts` does for you. Would that make sense from your point of view?

To clarify, a general infra overview is most helpful for users who need to set up their own infrastructure for a testnet / mainnet. These developers would have gone through the tutorial with `comit-scripts` and are now ready for 'the next phase' of bringing their application to production.

I think of the infra overview as an 'advanced' type of doc for the latter stages of a developer's journey to COMIT. For example, I imagine a COMIT developer would go through this journey:

- Arrive at comit.network
- Read about what COMIT does
- Read Getting Started and App Tutorial
- Read Core Concepts and Protocol explanation
- Finished going through COMIT App tutorial
- Developer is now ready to deploy their COMIT app
- **Infra Overview would be useful here**
- Developer has gone live with the COMIT App on testnet / mainnet

**Comment by D4nte** [comment]
I have update this issue so it just covers the "advance" part.

Because of that, it has low priority comparing to other doc items.

@bonomat I sent this issue to bottom of the pile.

**Comment by da-kami** [comment]
I think this is still valid and should remain open.
The development environment is tackled by https://github.com/comit-network/comit.network/pull/56
But moving to testnet should be added separately. 
I'd suggest to add a testnet section eventually where we could just mention the testnet-demo (so that devs are aware of it) - later we can do more...

**Comment by D4nte** [comment]
we still want to consider this at the moment
